library-core/src/main/java/com/mikepenz/fastadapter/items/SimpleItemList.java

```diff
@@ -0,0 +1,174 @@
+package com.mikepenz.fastadapter.items;
+
+import com.mikepenz.fastadapter.IItem;
+import com.mikepenz.fastadapter.adapters.IItemList;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+
+public class SimpleItemList<Item extends IItem> extends IItemList<Item> {
+  private final List<Item> list;
+
+  public SimpleItemList() {
+    this(-1);
+  }
+  
+  public SimpleItemList(int initialSize) {
+    list = initialSize < 0 ? new ArrayList<Item>() : new ArrayList<Item>(initialSize);
+  }
+  
+  @Override
+  public int size() {
+    return list.size();
+  }
+
+  @Override
+  public Item get(int position) {
+    return list.get(position);
+  }
+
+  @Override
+  public List<Item> getAll() {
+    return new ArrayList<>(list);
+  }
+
+  @Override
+  public void add(Item item) {
+    list.add(item);
+    if(listCallback != null) listCallback.onInserted(list.size() - 1, 1);

```

Comment:
An default list callback thats never null could be acceptable.

Reply:
Another idea I had was to throw an exception, because technically the user shouldn't be modifying the `IItemList` before "attaching" it to the adapter.

Reply:
Currently its possible to add items to the adapter before attaching it.

Reply:
That's fine. The concern here is if someone creates a custom `IItemList` and adds items to it before passing it to an adapter.